### ISSUE TITLE
Declarative pattern for response codes

### ISSUE THREAD

**Issue by dashby3000** [strongloop/strong-remoting#186]
In order to return a custom non-error response code, you must implement an afterRemoting hook.  The following is an example of an endpoint returning a status of 299.

```javascript
MyModel.remoteMethod(
   'foo',
   {
     description: 'Do some foo',
     accepts: [
     ],
     returns:{ root: true} ,
     http: {
       verb: 'GET',
       path: '/foo'
     },
     rest: {
       after: function(ctx, cb) {
         ctx.res.status(299);
         cb();
       }
     }
   }
 );
```

We request a more declarative approach.  Something like: `returns: { status: 299, root: true }`

ref: PLATAPI-234

This would be the mechanism for implementing https://github.com/strongloop/loopback/issues/360

**Comment by ritch** [comment]
Yeah this sounds good... It belongs in the `http` namespace though.

**Comment by bajtos** [comment]
Another option to consider: annotate a callback argument as holding the HTTP status code 

```js
function foo(cb) {
  cb(null, 299);
}
foo.returns = {
  arg: 'httpStatusCode', http: { target: 'statusCode' }
}
```

**Comment by ritch** [comment]
https://github.com/strongloop/strong-remoting/pull/218

**Comment by ritch** [comment]
Released as **v2.19.0**
